src/components/TextField/TextField.tsx

```diff
@@ -68,16 +81,30 @@ const TextField: React.FC<Props> = ({
   success = false,
   withErrorMsg = false,
   withIndicator = false,
+  size = DEFAULT_SIZE,
+  fill = DEFAULT_COLOR_CONFIG.fill,
+  fillShade = DEFAULT_COLOR_CONFIG.fillShade,
+  LeftIconWithSwatches,
   ...rest
 }) => {
+  const textFieldColors = generateTextFieldColors({ fill, fillShade });
+  const IconWrapper: FC = ({ children }) => (
+    <div css={iconWrapperStyle({ textFieldColors, rightIcon })}>{children}</div>
+  );
+
   return (
     <React.Fragment>
       <div css={{ display: 'flex', alignItems: 'center', position: 'relative' }}>
-        <div css={wrapperStyle({ disabled, error, lean, styleType })}>
-          <div css={textFieldStyle({ label, leftIcon })}>
-            {leftIcon && <div css={iconWrapperStyle({ label, rightIcon })}>{leftIcon}</div>}
+        <div css={wrapperStyle({ textFieldColors, disabled, error, lean, styleType })}>
+          <div css={textFieldStyle({ size, label, leftIcon, LeftIconWithSwatches })}>
+            {leftIcon && <IconWrapper>{leftIcon}</IconWrapper>}
+            {LeftIconWithSwatches && (
+              <IconWrapper>
+                <LeftIconWithSwatches colorConfig={{ fill, fillShade }} />
```

Comment:
why we pass this config to icon as we do have fill and shade and we can pass the actual color as color to the icon ?

Reply:
changed this